app/models/test_track/lazy_visitor_by_identity.rb

```diff
@@ -0,0 +1,38 @@
+class TestTrack::LazyVisitorByIdentity
```

Comment:
This model is a little interesting. `TestTrack::Visitor` objects will lazily fetch on demand, but always have an `id` attribute. When fetching a visitor by identity, that's no longer true - you actually fetch the visitor ID from TT so even retrieving `id` would require an API call.

So I added the concept of `id_loaded?` in addition to Visitor's existing concept of `loaded?`. As I said, regular visitors can always say `id_loaded => true`. But this lazy visitor will return `false` until it's actually fetched.

There are some housekeeping operations that aren't worth fetching `id` just to do them, but we'll gladly do them if the `id` is already loaded, so that's the rationale here.

For this model, `loaded?` and `id_loaded?` are the same, but for a regular `Visitor`, `loaded?` will return true only after the remote visitor has been fetched from the server to get the assignment bag.